vlm/src/main/scala/geotrellis/contrib/vlm/DataPath.scala

```diff
@@ -0,0 +1,15 @@
+package geotrellis.contrib.vlm
+
+import java.net.{URI, URL}
+
+
+trait DataPath {
+  def path: String
+  def targetPath: String
+
+  protected def servicePrefix: String
+
+  override def toString: String = targetPath
+  def toURI: URI = new URI(targetPath)
+  def toURL: URL = new URL(targetPath)
```

Comment:
We should avoid using `URL` in API because it's not serializable and binds to resources in surprising ways. This is "well known" java API issue.